src/Sulu/Bundle/AdminBundle/Resources/js/components/ColumnList/ToolbarDropdown.js

```diff
@@ -1,65 +1,61 @@
-import ToolbarDropdownList from './ToolbarDropdownList';
+import ArrowMenu from '../ArrowMenu';
 import type {ToolbarDropdown as ToolbarDropdownProps} from './types';
 import toolbarStyles from './toolbar.scss';
+import toolbarDropdownStyles from './toolbarDropdown.scss';
 
 @observer
 export default class ToolbarDropdown extends React.Component<ToolbarDropdownProps> {
     static defaultProps = {
         skin: 'primary',
     };
 
-    @observable popoverOpen: boolean = false;
-    @observable popoverAnchorElement: ?ElementRef<*>;
+    @observable open: boolean = false;
 
-    @action handleClick = (event: SyntheticEvent<HTMLOptionElement>) => {
-        this.popoverAnchorElement = event.currentTarget;
-        this.popoverOpen = true;
+    @action handleClick = () => {
+        this.open = true;
     };
 
-    @action handlePopoverClose = () => {
-        this.popoverOpen = false;
+    @action handleMenuClose = () => {
+        this.open = false;
     };
 
-    render() {
-        const {icon, options, skin} = this.props;
-
+    renderButton() {
+        const {icon, skin} = this.props;
         const className = classNames(
             toolbarStyles.item,
             toolbarStyles[skin]
         );
+        return (
+            <a className={className} onClick={this.handleClick}>
+                <Icon name={icon} />
+                <Icon className={toolbarDropdownStyles.buttonArrowIcon} name="su-angle-down" />
+            </a>
+        );
+    }
+
+    generateChildren() {
+        const {options} = this.props;
+        return options.map(({disabled, label, onClick}, index) => (
+            <ArrowMenu.Action disabled={disabled} key={index} onClick={onClick}>
+                {label}
+            </ArrowMenu.Action>
+        ));
+    }
 
+    render() {
         return (
             <Fragment>
-                <a className={className} onClick={this.handleClick}>
-                    <Icon name={icon} />
-                </a>
-                <Popover
-                    anchorElement={this.popoverAnchorElement}
-                    onClose={this.handlePopoverClose}
-                    open={this.popoverOpen}
-                >
-                    {
-                        (setPopoverElementRef, popoverStyle) => (
-                            <div
-                                ref={setPopoverElementRef}
-                                style={popoverStyle}
-                            >
-                                <ToolbarDropdownList
-                                    onClick={this.handlePopoverClose}
-                                    options={options}
-                                />
-                            </div>
-                        )
-                    }
-                </Popover>
+                <ArrowMenu anchorElement={this.renderButton()} onClose={this.handleMenuClose} open={this.open}>
+                    <ArrowMenu.Section>
+                        {this.generateChildren()}
```

Comment:
I prefer to have this inline, so that I don't have to jump around a lot of functions when reading this code. Especially in cases like this, where this would be quite easy. Instead of this line you can write something like this:

```javascript
{options.map(({label}) => (<ArrowMenu.Action ...>{label)</ArrowMenu.Action>))}
```